packages/salesforce/src/Adaptor.js

```diff
@@ -187,38 +196,37 @@ export function query(qs, options, callback = s => s) {
       throw err;
     }
 
-    if (qResult.totalSize > 0) {
-      console.log('Total records', qResult.totalSize);
-
-      while (!done) {
-        result.push(qResult);
-
-        if (qResult.done) {
-          done = true;
-        } else if (autoFetch) {
-          console.log(
-            'Fetched records so far',
-            result.map(ref => ref.records).flat().length
-          );
-          console.log('Fetching next records...');
-          try {
-            qResult = await connection.request({ url: qResult.nextRecordsUrl });
-          } catch (err) {
-            const { message, errorCode } = err;
-            console.log(`Error ${errorCode}: ${message}`);
-            throw err;
-          }
-        } else {
-          done = true;
+    const processRecords = async qResult => {
+      const { done, totalSize, records, nextRecordsUrl } = qResult;
+
+      result.done = done;
+      result.totalSize = totalSize;
+      result.records.push(...records);
+
+      if (!done && !autoFetch && nextRecordsUrl) {
+        result.nextRecordsUrl = nextRecordsUrl;
+      }
+      if (!done && autoFetch) {
+        console.log('Fetched records so far:', result.records.length);
+        console.log('Fetching next records...');
+
+        try {
+          const newResult = await connection.request({ url: nextRecordsUrl });
+          await processRecords(newResult);
+        } catch (err) {
+          const { message, errorCode } = err;
+          console.error(`Error ${errorCode}: ${message}`);
+          throw err;
         }
       }
+    };
 
-      console.log(
-        'Done ✔ retrieved records',
-        result.map(ref => ref.records).flat().length
-      );
+    if (qResult.totalSize > 0) {
+      console.log('Total records:', qResult.totalSize);
+      await processRecords(qResult);
+      console.log('Done ✔ retrieved records:', result.records.length);
     } else {
-      result.push(qResult);
+      result = qResult;
```

Comment:
Why this assignment? Is `qResult` ever any different than the result declaration above?

I know i'm being picky but I don't think it's a meaningful assignment, which means you can make result a `const` and the code just gets a bit clearer

Reply:
`{  done: true, totalSize: 0, records: [] }` is perfect for a result with no records. We don't care about `nextRecordsUrl`.

I don't see any need for re-assignment here

Reply:
But there are 0 records. We're specifically talking about the case where `qResult.totalSize === 0`.

We are overwriting the object `{ done: true, totalSize: 0, records: [] ` with  `{ done: true, totalSize: 0, records: [], nextRecordsUrl: ''}` . And it doesn't benefit anyone.

Reply:
Ooh i think i got it 💡, i have pushed an updated that removes the `qResult` reassignment thingy. Please have a look

Reply:
I don't understand that commit at all :joy_cat: :joy_cat: 

I'll sort this out and get this released